### PR TITLE
add "add-activity" functionality for collection-edit.

### PR DESCRIPTION
**Pull request by jlai-d2l**
Requires: https://github.com/BrightspaceHypermediaComponents/siren-sdk/pull/110

Screenshots
image
image

### CODE REVIEW

File: components/d2l-activity-collection-editor/d2l-activity-collection-editor.js [hunk 1]
```diff
@@ -74,12 +80,72 @@ class CollectionEditor extends EntityMixinLit(LitElement) {
...
+		if (clearList) {
+			this._candidateItems = [];
+		}
+		this._actionCollectionEntity._items().forEach(item => {
+			item.onActivityUsageChange(async usage => {
+				usage.onOrganizationChange((organization) => {
+					const alreadyAdded = this._items.findIndex(item => item.self() === organization.self()) >= 0;
+					this._candidateItems.push({ item, organization, alreadyAdded });
+					if (this._candidateItems.length >= this._actionCollectionEntity._items().length) {
+						this.requestUpdate('_candidateItems', []);
```
Comment (m6jones) [inline components/d2l-activity-collection-editor/d2l-activity-collection-editor.js]: I'd prefer if you used the promise thing that way we can handle errors later. But really, I can change it when we actually handle the errors. 

Ie, make the choice if that is actually better when we need it.

File: components/d2l-activity-collection-editor/d2l-activity-collection-editor.js [hunk 2]
```diff
@@ -74,12 +80,72 @@ class CollectionEditor extends EntityMixinLit(LitElement) {
...
+	async open() {
+		this.getCandidates(this._addExistingAction, null, true);
+		await this.shadowRoot.querySelector('d2l-dialog').open();
+	}
+
 	static get properties() {
 		return {
 			_description: { type: String },
 			_items: { type: Array },
+			_candidateItems: { type: Array },
```
Comment (m6jones) [inline components/d2l-activity-collection-editor/d2l-activity-collection-editor.js]: NIT: I like this list sorted by alpha

File: components/d2l-activity-collection-editor/d2l-activity-collection-editor.js [hunk 3]
```diff
@@ -121,6 +187,26 @@ class CollectionEditor extends EntityMixinLit(LitElement) {
 				align-self: flex-start;
 				margin: 0.55rem 1.5rem;
 			}
+			.d2l-add-activity-dialog {
+				min-height: 500px;
+			}
+			.d2l-add-activity-dialog-header {
+				display: flex;
```
Comment (m6jones) [inline components/d2l-activity-collection-editor/d2l-activity-collection-editor.js]: Alpha sort the css

File: components/d2l-activity-collection-editor/d2l-activity-collection-editor.js (hunk 1 above)
Comment (jlai-d2l) [reply components/d2l-activity-collection-editor/d2l-activity-collection-editor.js]: I found that the `subEntitiesLoaded()` didn't work for me, as the I had to manually create the collection (ActionCollectionEntity), and as it didn't go through `_subEntity()`, there was no aggregation of promises to know when everything had loaded. Perhaps I wasn't using it correctly, but I found that this worked, and was able to unblock me. I'd be interested in seeing your approach when we have time to come back and look at this.

**Pull request merged**